### PR TITLE
Separate basic chemicals into HVC, chlorine, methanol and ammonia

### PR DESCRIPTION
**Pull request by nworbmot**
The category of "basic chemicals" comes from the JRC IDEES database. It consists to our knowledge of high value chemicals (HVC), chlorine, methanol and ammonia. HVC are used for plastics synthesis and consist of ethylene, propylene and BTX (benzene, toluene, and the three xylene isomers).

It is necessary to separate out these chemicals because current and future production routes are different. For example, the hydrogen necessary for ammonia synthesis could be replace by green hydrogen. As another important example, primary HVC production could be reduced by mechanical and chemical recycling of plastics.

Previously ammonia production had already been separated from basic chemicals.

In this pull request (PR) in addition methanol and chlorine are also separated, based on energy consumption data from (DECHEMA, 2017). What is left is HVC production. The resulting specific energy demand for HVC in ``resources/industry_sector_ratios.csv`` (2.9 MWh_elec/tHVC, 4 MWh_CH4/tHVC, 0.2 MWh_heat/tHVC, 14 MWh_naphtha/tHVC) is relatively close to the numbers quoted in (Lechtenböhmer et al, 2016) (4.7 MWh_misc_energy/tHVC, 14 MWh_naphtha_feedstock/tHVC).

In addition, the option to specify mechanical and chemical recycling of plastics is given. This reduces the primary HVC production, and has a dramatic effect on costs. Assuming plastics demand stays constant, in a net-zero-CO2-emission scenario for Europe, producing 100% of HVC via primary route is 73 bnEUR/a or 14% more expensive than a scenario with 20% primary production, 30% mechanical recycling, 10% chemical recycling and 40% reuse (reuse has no energy use). A reasonable circular economy scenario, following Material Economics (2019) page 133 could be e.g. 25-25-25-25% for these different options.

What about process emissions from recycling plastics? In the current model setup, this is taken care of in that all carbon in primary production lands in the atmosphere. Suppose ``x+y+z`` carbon units of plastics are produced in a year, ``x`` primary, ``y`` mechanical and ``z`` chemical. Feedstocks required: ``x(1+c)`` primary (feedstock: naphtha), ``y(1+d)`` mechanical (feedstock: waste plastics), ``z(1+e)`` chemical (feedstock: waste plastics). There are process emissions of ``xc`` for primary production, ``yd`` for mechanical recycling and ``ze`` for chemical recycling. Of the ``x+y+z`` units in produced plastics, ``y(1+d)+z(1+e)`` are recycled for next year and ``x - yd - ze`` are burned in waste-to-energy plants. Thus total emissions are ``xc + yd + ze + (x - yd - ze) = x(1+c)``. The model currently allows the ``xc`` process emissions to be captured and the ``x`` goes into the atmosphere, correctly taking account of everything. We could allow process emissions from mechanical and chemical recycling to be captured.

Some loose ends that could be fixed:
- [ ] Process emissions from methanol production are ignored and are all landing at HVC. Could separate: (DECHEMA, 2017) has 1.5 tCO2/tMeOH with process emissions from feedstocks, i.e. originating from CH4 input. It's a very small effect. #171 
- [ ] Allow process emissions from mechanical and chemical recycling to be captured. These need to be specified first; at the moment they are landing in the atmosphere. #170 
- [ ] The separation of basic chemicals hasn't been applied to the scripts for "today's industry energy consumption". This is not necessary for future scenarios, but would be useful for future pathway investigations. #172 
- [x] We need a reliable source for the energy demand for mechanical recycling. (Accenture, 2017), cited by (DECHEMA, 2017), has on page 35 something approaching 7 MWh/tHVC, which seems high, around same as chemical recycling.
- [ ] Adding waste-to-energy (WtE) plants to consume non-recycled plastics (``x - yd - ze`` in above). I've already added this in a test, but the effect is small. WtE plants are expensive and only built if waste is forced to be used. The model chooses unabated WtE over WtE with carbon capture. Could argue this is outside system boundary and continue assuming all carbon in primary plastics lands in atmosphere. #173 

References:
- Lechtenböhmer et al, 2016: [Decarbonising the energy intensive basic materials industry through electriﬁcation - Implications for future EU electricity demand](http://dx.doi.org/10.1016/j.energy.2016.07.110)
- DECHEMA, 2017: [Low carbon energy and feedstock for the European chemical industry](https://dechema.de/dechema_media/Downloads/Positionspapiere/Technology_study_Low_carbon_energy_and_feedstock_for_the_European_chemical_industry-p-20002750.pdf)
- Accenture (2017): [Taking the European Chemical Industry into the Circular Economy](https://cefic.org/app/uploads/2019/02/Accenture-Cefic-circular-economy-brochure.pdf)
- Material Economics (2019): [Industrial Transformation 2050 - Pathways to Net-Zero Emissions from EU Heavy Industry](https://materialeconomics.com/latest-updates/industrial-transformation-2050)

### CODE REVIEW

File: config.default.yaml [hunk 1]
```diff
@@ -270,10 +270,23 @@ industry:
   MWh_elec_per_tNH3_electrolysis: 1.17 # from https://doi.org/10.1016/j.joule.2018.04.017 Table 13 (air separation and HB)
   NH3_process_emissions: 24.5 # in MtCO2/a from SMR for H2 production for NH3 from UNFCCC for 2015 for EU28
   petrochemical_process_emissions: 25.5 # in MtCO2/a for petrochemical and other from UNFCCC for 2015 for EU28
-  HVC_primary_fraction: 1.0 #fraction of current non-ammonia basic chemicals produced via primary route
+  HVC_primary_fraction: 1. # fraction of today's HVC produced via primary route
+  HVC_mechanical_recycling_fraction: 0. # fraction of today's HVC produced via mechanical recycling
+  HVC_chemical_recycling_fraction: 0. # fraction of today's HVC produced via chemical recycling
+  HVC_production_today: 52. # MtHVC/a from DECHEMA (2017), Figure 16, page 107; includes ethylene, propylene and BTX
+  MWh_elec_per_tHVC_mechanical_recycling: 3. # estimate
```
Comment (fneum) [inline config.default.yaml:277]:
```suggestion
  MWh_elec_per_tHVC_mechanical_recycling: 0.547 # from SI of https://doi.org/10.1016/j.resconrec.2020.105010, Table S5, for HDPE, PP, PS, PET. LDPE would be 0.756.
```
Adds the energy use of mechanical recycling. 

I have assumed energy demand for recycling of HPTE, PET, PS, PP as they are all very similar whereas LDPE is higher as well as more difficult to recycle.

    
    https://www.generalkinematics.com/blog/different-types-plastics-recycled/

File: config.default.yaml [hunk 2]
```diff
@@ -270,10 +270,23 @@ industry:
   MWh_elec_per_tNH3_electrolysis: 1.17 # from https://doi.org/10.1016/j.joule.2018.04.017 Table 13 (air separation and HB)
   NH3_process_emissions: 24.5 # in MtCO2/a from SMR for H2 production for NH3 from UNFCCC for 2015 for EU28
   petrochemical_process_emissions: 25.5 # in MtCO2/a for petrochemical and other from UNFCCC for 2015 for EU28
-  HVC_primary_fraction: 1.0 #fraction of current non-ammonia basic chemicals produced via primary route
+  HVC_primary_fraction: 1. # fraction of today's HVC produced via primary route
+  HVC_mechanical_recycling_fraction: 0. # fraction of today's HVC produced via mechanical recycling
+  HVC_chemical_recycling_fraction: 0. # fraction of today's HVC produced via chemical recycling
```
Comment (fneum) [inline config.default.yaml:275]: Should we set defaults for recycling where not all plastics are primary?

E.g. mechanical (20-30%), chemical (15%), reuse (5-10%) recycling.

Long-term it should be possible to set an exogenous pathway.

In the current implementation, would reuse (no energy demand) be the 1 - primary - mechanical - chemical?

File: scripts/build_industrial_production_per_country.py [hunk 3]
```diff
@@ -198,9 +198,16 @@ def add_ammonia_demand_separately(demand):
     # EE, HR and LT got negative demand through subtraction - poor data
     demand['Basic chemicals'].clip(lower=0., inplace=True)
 
-    to_rename = {"Basic chemicals": "Basic chemicals (without ammonia)"}
-    demand.rename(columns=to_rename, inplace=True)
+    demand.insert(2, "HVC", 0.)
+    demand.insert(3, "Chlorine", 0.)
+    demand.insert(4, "Methanol", 0.)
```
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:203]: Only necessary for nice ordering of df.

File: scripts/build_industrial_production_per_country.py [hunk 4]
```diff
@@ -198,9 +198,16 @@ def add_ammonia_demand_separately(demand):
...
 
+    # assume HVC, methanol, chlorine production proportional to non-ammonia basic chemicals
+    demand["HVC"] = config["HVC_production_today"]*1e3/demand["Basic chemicals"].sum()*demand["Basic chemicals"]
+    demand["Chlorine"] = config["chlorine_production_today"]*1e3/demand["Basic chemicals"].sum()*demand["Basic chemicals"]
+    demand["Methanol"] = config["methanol_production_today"]*1e3/demand["Basic chemicals"].sum()*demand["Basic chemicals"]
```
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:208]:
```suggestion
    distribution_key = demand["Basic chemicals"] / demand["Basic chemicals"].sum()
    demand["HVC"] = config["HVC_production_today"] * 1e3 * distribution_key
    demand["Chlorine"] = config["chlorine_production_today"] * 1e3 * distribution_key
    demand["Methanol"] = config["methanol_production_today"] * 1e3 * distribution_key
```

File: scripts/build_industrial_production_per_country.py [hunk 5]
```diff
@@ -179,8 +179,8 @@ def industry_production(countries):
     return demand
 
 
-def add_ammonia_demand_separately(demand):
-    """Include ammonia demand separately and remove ammonia from basic chemicals."""
+def separate_basic_chemicals(demand):
+    """Remove ammonia, chlorine and methanol from basic chemicals to get HVC."""
```
Comment (fneum) [inline scripts/build_industrial_production_per_country.py:183]:
```suggestion
    """Separate basic chemicals into ammonia, chlorine, methanol and HVC."""
```
HVC is not the rest in the current implementation, but has its own input in the `config.yaml`.

File: scripts/build_industrial_production_per_country_tomorrow.py [hunk 6]
```diff
@@ -39,11 +39,14 @@
...
     al_primary_fraction = get(config["Al_primary_fraction"], investment_year)
     fraction_persistent_primary = al_primary_fraction * total_aluminium.sum() / production[key_pri].sum()
-    
+
     production[key_pri] = fraction_persistent_primary * production[key_pri]
     production[key_sec] = total_aluminium - production[key_pri]
 
-    production["Basic chemicals (without ammonia)"] *= config['HVC_primary_fraction']
+    production.insert(4, "HVC (mechanical recycling)", config["HVC_mechanical_recycling_fraction"]*production["HVC"])
+    production.insert(5, "HVC (chemical recycling)", config["HVC_chemical_recycling_fraction"]*production["HVC"])
```
Comment (fneum) [inline scripts/build_industrial_production_per_country_tomorrow.py:47]:
```suggestion
    production["HVC (mechanical recycling)"] = config["HVC_mechanical_recycling_fraction"] * production["HVC"])
    production["HVC (chemical recycling)"] = config["HVC_chemical_recycling_fraction"] * production["HVC"])
```
Question is if we want to manually maintain the ordering.

File: config.default.yaml (hunk 1 above)
Comment (nworbmot) [reply config.default.yaml:277]: Sounds good! Also for reference, another source [doi:10.1016/j.scitotenv.2017.05.278](http://dx.doi.org/10.1016/j.scitotenv.2017.05.278) with somewhat lower energy demand around 0.3 MWh/tPlastic. We can take the newer higher number.

File: config.default.yaml (hunk 2 above)
Comment (nworbmot) [reply config.default.yaml:275]: Yes, that's a good idea, but probably it should be a path 2020-50 with increasing rates of recycling and reuse.

File: scripts/build_industry_sector_ratios.py [hunk 7]
```diff
@@ -374,52 +374,84 @@ def chemicals_industry():
...
-    assert sector in str(s_out.index)
-
-    ammonia = pd.read_csv(snakemake.input.ammonia_production, index_col=0)
-
-    # ktNH3/a
-    ammonia_total = ammonia.loc[ammonia.index.intersection(eu28), str(year)].sum()
 
-    s_out -= ammonia_total
+    # convert from MtHVC/a to ktHVC/a
+    s_out = config["HVC_production_today"]*1e3
```
Comment (fneum) [inline scripts/build_industry_sector_ratios.py:380]: Previously, `s_out` included also chlorine and methanol. Is using only HVC here correct?

File: config.default.yaml (hunk 1 above)
Comment (nworbmot) [reply config.default.yaml:277]: Could we also check that the numbers from https://doi.org/10.1016/j.resconrec.2020.105010 are consistent with our chemical recycling assumptions? @brynpickering warned about reliability of Material Economics numbers in a private email.

**Comment by fneum** [comment]
Resulting change in overall energy demands (2050):

|                                 |   old |   new |
|:--------------------------------|------:|------:|
| electricity                     |  1697 |  1721 |
| coal                            |     0 |     0 |
| coke                            |     0 |     0 |
| solid biomass                   |   702 |   702 |
| methane                         |   380 |   381 |
| hydrogen                        |   204 |   195 |
| low-temperature heat            |    59 |    59 |
| naphtha                         |   778 |   728 |
| process emission                |   127 |   127 |
| process emission from feedstock |    27 |    25 |
| current electricity             |  1040 |  1040 |

Little more electricity, little less hydrogen, less naphtha.

File: config.default.yaml (hunk 1 above)
Comment (fneum) [reply config.default.yaml:277]: The paper describes for routes of chemical recycling:

1) refinery feedstock: liquefy plastic -> use as a substitute for oil or naphtha in refinery and steam cracking processes
2) fuel production: gasification / pyrolysis -> gaseous / liquid fuels
3) monomer production: polyolefins -> monomers (e.g. ethylene) through thermal/catalytic pyrolysis
4) chemical upcycling

Route 3) comes closes to the Material Economics study (which uses pyrolysis and electric steam cracking and methanol synthesis to produce HVC from HVC for which they need 6.9 kWh/kg **electricity** with a yield of 0.91 kg/kg).

Route 3) in the paper following table S10 in the SI requires 3 MJ/kg (0.78 kWh/kg) of **heat** to turn HDPE into ethylene with a yield of 0.85 kg_ethylene / kg.

Not sure how conclusive this comparison is. Seems Material Economics numbers are a bit high?

*polyolefins: LDPE, HDPE, PP
*HVC: ethylene, propylene, polyolefins etc.

**Comment by fneum** [comment]
For `HVC_primary_fraction`, `HVC_mechanical_recycling_fraction` and `HVC_chemical_recycling_fraction` you can now also set a pathway, e.g.:

```yaml
HVC_primary_fraction:
    2020: 1.0
    2030: 0.8
    2040: 0.5
    2050: 0.0

**Pull request merged**